cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ZookeeperScaler.java

```diff
@@ -99,9 +99,16 @@ protected ZookeeperScaler(Vertx vertx, ZooKeeperAdminProvider zooAdminProvider,
 
                     getCurrentConfig(zkAdmin)
                             .compose(servers -> scaleTo(zkAdmin, servers, scaleTo))
-                            .compose(nothing -> closeConnection(zkAdmin))
-                            .onSuccess(scalePromise::complete)
-                            .onFailure(scalePromise::fail);
+                            .onComplete(res ->
+                                closeConnection(zkAdmin)
+                                    .onComplete(closeResult -> {
```

Comment:
Should you log something in case `closeResult` fails? I think it might be useful.

Reply:
Ok, I'll add that. Do you think debug or info (or even warn) level?

Reply:
I think that unless we know for sure that any issues will print other errors from the Zoo client, then `warn` level might be good.

Reply:
It's actually already logging exceptions from `ZooKeeperAdmin#close` in `closeConnection`. Maybe just bumping that up from `debug` to `warn` will do it?

Reply:
Ok, that makes sense I guess.

Reply:
Done. That will cover failures here as well as any that may occur while closing from `connect`.